### ISSUE TITLE
I'm writing a log file analyser and want to integrate LSM infos

### ISSUE THREAD

**Issue by originalfoo** [thale5/LSM#1]
Hi,

I'm writing a log file analyser that uses modular parsers to process info in the log file and create more meaningful info.

WIP screenshot showing some of the categories already added:

![log](https://user-images.githubusercontent.com/1386719/54189213-39b78100-44a9-11e9-86c9-f8361270f61a.png)

I'd like to fully integrate the LSM log file entries in to my analyser, possibly even rendering graphs of the load process.

I was wondering if you could provide more details on some of the following log entries to save me having to dig through the code (much of which I don't understand)...

I'll just dump the types of entries I'm aware of below and hopefully you can explain them? :D

```
[LSMT] Options: 72 True True True True True True True

[LSMT] Residential Low: -> syntax error

[LSMT] Empty Park 12x8 -> skipped

[LSMT] Pedestrian Elevated Pillar -> not skipped because excepted

[LSMT] mainThreadQueue len 1 at 75982

[LSMT] Skipped 606 building prefabs

[LSMT] GetLoadQueue 76014

[LSMT] Sorted at 76078

[LSMT] Duplicate name 1574853410.r-pillar-s_Data

[LSMT] LoadQueue 2920 77709

[LSMT] 0: 77710 0 0 1084831728.Long Grass Light Green _Data

[LSMT] 1: 78041 1 24 1084831728.Long Grass Dark Green_Data

[LSMT] Mem 77711 4633 8482 0 0

[LSMT] Unsupported type for deserialization: NetInfo

[LSMT] Asset missing: 513647597.Eating Cow (static)_Data

[LSMT] Textures / Materials / Meshes shared: 6104 / 2166 / 4068 pre-loaded: 10166 / 6700 / 4079 loaded: 2 / 75 / 0

(for above, what is difference between pre-loaded and loaded?)

[LSMT] Max cache 291
```

**Comment by originalfoo** [comment]
Deduced by looking at code

**Comment by thale5** [comment]
Sorry, I missed your request earlier and noticed it just now.
I use those log lines to analyse loading issues and improve loading performance. I suppose most of them are quite irrelevant for anyone else than me. What is more, they have changed and will probably change over time.
This line might be of interest:
[LSMT] Textures / Materials / Meshes shared: 6104 / 2166 / 4068 pre-loaded: 10166 / 6700 / 4079 loaded: 2 / 75 / 0'Shared' refers to LSM resource sharing. It is the fastest and most memory-efficient way to load resources: the main thread re-uses a resource it has seen before.
'Pre-loaded' is the 2nd best way to load resources. It means the LSM loader thread was able to queue and load the resource for the main thread, which then prepares the resource for use.
'Loaded' is the worst alternative: the main thread had to load and prepare the resource.

    On Saturday, May 4, 2019, 5:44:53 PM GMT+3, Aubergine <notifications@github.com> wrote:  
 
 
Closed #1.

**Comment by originalfoo** [comment]
Thanks for the info! I was wondering if it would be possible to ensure a consistent output of memory consumption (just main RAM is fine, assuming that is possible also on Mac and Linux)?

I was hoping to use this sort of thing to create a graph of memory over time:
```
[LSMT] Mem 77711 4633 8482 0 0
```

I am able to parse that line if it appears, however it doesn't seem to appear in most log files, which is a shame.
